lib/socket.ex

```diff
@@ -192,8 +218,28 @@ defmodule Kadabra.Socket do
   end
 
   def handle_info({:ssl_closed, socket}, state) do
-    :telemetry.execute([:kadabra, :socket, :closed], %{}, %{type: :ssl, socket: socket})
+    :telemetry.execute([:kadabra, :socket, :closed], %{}, %{
+      type: :ssl,
+      socket: socket
+    })
+
     Kernel.send(state.active_user, {:closed, self()})
     {:noreply, %{state | socket: nil}}
   end
+
+  defp get_peername(pid = {:ssl, _, _}) do
+    case :ssl.peername(pid) do
+      {:ok, peername} -> peername
+      _ -> nil
+    end
+  end
+
+  defp get_peername({_, {_, pid, _, _}, _}) do
```

Comment:
I thought we could just call this on the socket itself if the socket wasn't an ssl socket but the one returned by `:gen_tcp.connect`. But then again, I can't find exactly in the `:gen_tcp` docs where they describe the structure of this socket, but I didn't think it had this shape.

Reply:
Yeah I'm really not sure about this case. It's probably wrong, but I'll deal with it if/when it becomes an issue.